### ISSUE TITLE
Add support for no javascript and keyboard navigation

### ISSUE THREAD

**Issue by iacore** [tauri-apps/tauri-docs#307]
Currently, the documentation is unusable without Javascript. For example, on page https://tauri.studio/en/docs/api/cli, the side bar / links are useless when clicked.

Also, when navigating with keyboard, <kbd>Space</kbd> on "Skip to content" does not scroll the page to start of content (it should be a link with `href="#start"`). Navigating backwards is confusing, since the title bar has `display: absolute` or something, and covers the first few lines of content, so I can't see the element in focus. Try Shift+Tab on the above page and you will understand.

**Comment by nothingismagick** [comment]
this is not something we can fix afaik - because we use docusaurus. sorry :(

**Comment by iacore** [comment]
The docusaurus documentation is usable without Javascript. https://docusaurus.io/docs

Here's a better reference of sidebar navigation: https://www.typescriptlang.org/docs/handbook/advanced-types.html

**Comment by nothingismagick** [comment]
Oh wow! I didn't know that!

**Comment by lorenzolewis** [comment]
@locriacyber can you test the navigation sidebar on this test deploy to see if the issue is fixed for you: https://deploy-preview-314--tauri.netlify.app/en/docs/api/cli/

Could you elaborate a bit more on the navigating with keyboard point? Give it a try on the above site as well as it could possibly be fixed.

**Comment by iacore** [comment]
- The "Skip to main content" link still needs JavaScript (incorrect `a[href]`).
- links (`<a>`) do not change color or anything on focus. I don't know which link I'm focusing.
- link color too light, too-little contrast with white background
- nav bar covers main content (please don't use `position: sticky`); Github uses this feature as well, but in Firefox, the browser considers text covered by sticky elements "visible".

![image](https://user-images.githubusercontent.com/74560659/149811629-de86e2e6-52ea-496d-b759-c6f3626ba79f.png)

**Comment by lorenzolewis** [comment]
> The "Skip to main content" link still needs JavaScript (incorrect a[href]).

> nav bar covers main content (please don't use position: sticky); Github uses this feature as well, but in Firefox, the browser considers text covered by sticky elements "visible".

Out of curiosity, can you test these two points on https://docusaurus.io/docs with the same setup? If it has the same behavior there then I'm afraid that's something upstream that we can't do anything about unless they adjust.

For the overall contrast, we're working on a redesign and I'll add the other two styling points to get covered under that

**Comment by iacore** [comment]
It's same behavior with "Jump to main content" and `:focus` style.

**Comment by lorenzolewis** [comment]
But the navbar covering the main content is fine there? I can compare between and see what we can do to fix that. Thanks for checking!

**Comment by slorber** [comment]
Those are Docusaurus issues that we'll try to fix, I don't think Tauri should do anything about those

**Comment by lorenzolewis** [comment]
@slorber feel free to reach out if any testing needs to be done around that when ready

**Comment by Josh-Cena** [comment]
Hey, for the theme colors part, we have included a11y guidelines in our styling docs: https://docusaurus.io/docs/next/styling-layout#styling-your-site-with-infima You can use the tool to select different light/dark mode palettes, and check if they pass the WCAG contrast check.

**Comment by lorenzolewis** [comment]
Thanks @Josh-Cena, we have a redesign in the works so we'll be paying special attention to light themes which should address this. 

@locriacyber I've included these notes in https://github.com/tauri-apps/tauri-docs/pull/333 so that we include them. I'm going to go ahead and close this issue for now. Any updates we get from upstream will be included when they are released.
